Tasks/PowerShellOnTargetMachines2/JobHelper.ps1

```diff
@@ -0,0 +1,33 @@
+function Run-RemoteScriptJobs {
+    [CmdletBinding()]
+    Param(
+        [System.Management.Automation.Runspaces.PSSession[]] $sessions,
+        [scriptblock] $script,
+        [System.Array] $scriptArguments,
+        [switch] $ignoreLASTEXITCODE
+    )
+    Trace-VstsEnteringInvocation -InvocationInfo $MyInvocation -Parameter "sessions","scriptArguments"
+    try {
+        $jobs = Invoke-Command -Session $sessions -AsJob -ScriptBlock $script -ArgumentList $scriptArguments
+        $jobResults = @();
+    
+        while($jobs | Where-Object { $_.State -eq "Running"}) {
+            ForEach($job in $jobs) {
+                Receive-Job -Job $job | ForEach-Object { if($_.VstsTask -eq $true) { $jobResults += $_ } else { Write-Host $($_ | Out-String) } }
+            }
+            Start-Sleep -Seconds 30
+        }
```

Comment:
Can you test appending $_ to jobResults with outputs that are non string type.

Reply:
Why do you need a sleep of 30 seconds?

Reply:
yes, this will work. jobresult is a hashtable and is received as is.

Reply:
a continuous looping here without a delay would cause a lot of interleaving of output from different machines.